### PR TITLE
[FEATURE] More semantic error codes on access check

### PR DESCRIPTION
**Pull request by e0ipso**
If a request is denied access via a RESTful exception do not discard the error code. Instead return a well formatted error response.

### CODE REVIEW

**Comment by e0ipso** [comment]
This PR supersedes #752.

File: tests/RestfulRenderCacheTestCase.test [hunk 1]
```diff
@@ -245,7 +247,7 @@ class RestfulRenderCacheTestCase extends \RestfulCurlBaseTestCase {
     // Try to access the cached resource.
     $response = $this->httpRequest($path);
     // The user should get a 403.
-    $this->assertEqual($response['code'], 403, 'Access denied for anonymous user.');
+    $this->assertEqual($response['code'], 401, 'Access denied for anonymous user.');
```
Comment (amitaibu) [inline tests/RestfulRenderCacheTestCase.test:250]: why 401? Also the comment above still says 403

Comment (e0ipso) [inline tests/RestfulRenderCacheTestCase.test:250]: Yikes. I'll fix the comment.

401 means authorization error, meaning that the requester meant to be identified with a Drupal user but the credentials were bad.

403 is more for, _I know who you are, and you cannot do that_.

Comment (amitaibu) [inline tests/RestfulRenderCacheTestCase.test:250]: sure, just the comment made me think it was wrong :)

**Comment by e0ipso** [comment]
This was merged.

Thanks for the effort!

**Comment by ademarco** [comment]
Great! Thanks!

**Pull request closed without merging**